backend/src/middleware/permissionsMiddleware.js

```diff
@@ -70,6 +70,29 @@ const onlyEnabledContent = rule({
   return !(disabled || deleted)
 })
 
+const invitationLimitReached = rule({
+  cache: 'no_cache',
+})(async (parent, args, { user, driver }) => {
+  const session = driver.session()
+  try {
+    const result = await session.run(
+      `
+      MATCH (user:User {id:$id})-[:GENERATED]->(i:InvitationCode)
+      RETURN COUNT(i) >= 3 as limitReached
+      `,
+      { id: user.id },
+    )
+    const [limitReached] = result.records.map(record => {
+      return record.get('limitReached')
+    })
+    return limitReached
+  } catch (e) {
+    throw e
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-04T07:15:54Z" title="Thursday, July 4th 2019, 9:15:54 am +02:00">Jul 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -70,6 +70,29 @@ const onlyEnabledContent = rule({
   return !(disabled || deleted)
 })
 
+const invitationLimitReached = rule({
+  cache: 'no_cache',
+})(async (parent, args, { user, driver }) => {
+  const session = driver.session()
+  try {
+    const result = await session.run(
+      `
+      MATCH (user:User {id:$id})-[:GENERATED]->(i:InvitationCode)
+      RETURN COUNT(i) >= 3 as limitReached
+      `,
+      { id: user.id },
+    )
+    const [limitReached] = result.records.map(record => {
+      return record.get('limitReached')
+    })
+    return limitReached
+  } catch (e) {
+    throw e
```


Would be more saying to have `err` here instead of `e`.
We had that discussion in other places already …

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-04T10:13:58Z" title="Thursday, July 4th 2019, 12:13:58 pm +02:00">Jul 4, 2019</time>_

---


I tend to agree, or even `error`, what would we lose have two extra characters, however, I think people who are somewhat familiar with JavaScript will know what this means.

I feel like there is a fine line between allowing for different coding styles and adopting community standards that we enforce. Maybe we could talk in a `Human Connection Developer's Community Meeting`?